src/PasswordHasher/PasswordHasherFactory.php

```diff
@@ -49,7 +49,7 @@ public static function build($passwordHasher)
 
         $hasher = new $className($config);
         if (!($hasher instanceof PasswordHasherInterface)) {
-            throw new RuntimeException('Password hasher must extend AbstractPasswordHasher class.');
+            throw new RuntimeException('Password hasher must implement PasswordHasherInterface.');
```

Comment:
Hashers aren't strictly related to authentication? You'll need them when creating user credentials as well. This can happen via API service, shell, or "normal" form POST by an user.

Are the hashers in any way coupled with the identifiers? I don't think so. So IMHO it's OK if they stay in their own namespace as I don't see them being a "subordinate" of authentication.